packages/server/lib/xhr_ws_server.ts

```diff
@@ -11,55 +11,76 @@ function trunc (str) {
   })
 }
 
+type DeferredPromise<T> = {
+  resolve: Function
+  reject: Function
+  p: Bluebird<T>
```

Comment:
call this `promise` which mimics the existing `Bluebird.defer()` interface and is consistent with how we use this in other places.

![image](https://user-images.githubusercontent.com/1268976/67883561-afe12480-fb1a-11e9-8e95-f7c563c14b35.png)